arkit/server.py

```diff
@@ -0,0 +1,256 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License in the file LICENSE.txt or at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This server will communicate with the AR app with websocket.
+cozmo will try 3 times, it controls by the round number self._round (0-2)
+1. see fireworks, surprise
+2. see fireworks, fail launch
+3. see fireworks, super happy and excited
+"""
+import time
+import websockets
+import asyncio
+import threading
+import asyncio
+import sys
+import PIL.Image
+import base64
+import os
+import io
+import random
+import cozmo
+from cozmo.util import degrees, distance_mm, speed_mmps, Pose
+from cozmo.objects import CustomObject, CustomObjectMarkers, CustomObjectTypes
+from lib.vision_cube import VisionCube
+from lib.config import IP_ADDRESS, PORT
+
+stopFlag = False
+
+USE_VIEWER = False
+USE_LOGGING = False
+DEBUG_MODE = True
+PURPLE = cozmo.lights.Color(rgb=(141, 7, 255))
+BLUE = cozmo.lights.Color(rgb=(6, 6, 255))
+YELLOW= cozmo.lights.Color(rgb=(255, 255, 7))
+GREEN = cozmo.lights.Color(rgb=(0, 255, 0))
+ORANGE = cozmo.lights.Color(rgb=(255, 177, 6))
+PINK = cozmo.lights.Color(rgb=(255, 13, 144))
+
+serverT = threading.Thread()
+
+class TapCube(threading.Thread):
+
+    def __init__(self):
+        threading.Thread.__init__(self)
+        self._round = 0
+        self._cube = 0
+        self._robot = None
+        self._wait_for_tap_flag = False
+        self._did_cozmo_tap_flag = False
+        self._tap_action_finished = False
+        self._animation_triggerd = False
+        self._trigger_message = False
+        self._message = ""
+
+        if USE_LOGGING:
+            cozmo.setup_basic_logging()
+        if USE_VIEWER:
+            cozmo.connect_with_tkviewer(self.run)
+
+        else:
+            try:
+                cozmo.connect(self.run)
+            except cozmo.ConnectionError as e:
+                sys.exit("A connection error occurred: %s" % e)
+
+    async def run(self, coz_conn):
+        """The run method runs once Cozmo is connected."""
+        await self.set_up_cozmo(coz_conn)
+
+        while True:
+            await asyncio.sleep(0)
+        pass
+
+    async def set_up_cozmo(self, coz_conn):
+        asyncio.set_event_loop(coz_conn._loop)
+        self._robot = await coz_conn.wait_for_robot()
+
+        global serverT
+        serverT = threading.Timer(0, self.run_server).start()
+        await self._robot.set_lift_height(0, duration=0.5).wait_for_completed()
+        await self._robot.play_anim("anim_launch_wakeup_02").wait_for_completed()
```

Comment:
We try to avoid referring to anims directly by name in publicly released scripts, and instead recommend using the animation triggers, I suspect this might not be fixable now, but the issue is that this makes it very likely that the script will fail on a future app version.